### PR TITLE
Performing JSON -> object conversions for Javascript input bindings

### PR DESCRIPTION
**Pull request by mathewc**
Addresses https://github.com/Azure/azure-webjobs-sdk-script/issues/154

### CODE REVIEW

File: src/WebJobs.Script/Description/NodeFunctionInvoker.cs [hunk 1]
```diff
@@ -280,12 +283,8 @@ protected override void OnScriptFileChanged(object sender, FileSystemEventArgs e
             Type triggerParameterType = input.GetType();
             if (triggerParameterType == typeof(string))
```
Comment (davidebbo) [inline src/WebJobs.Script/Description/NodeFunctionInvoker.cs]: I know that was there before, but why do type equality instead if `is` check on the `input` object?

**Comment by davidebbo** [comment]
Other than nit about previously existing code, looks good. Consider approved. :shipit:

**Comment by brettsam** [comment]
Looks awesome. I was about to send a PR that still uses JSON.parse -- I'll wait for this and rebase.

File: src/WebJobs.Script/Description/NodeFunctionInvoker.cs (hunk 1 above)
Comment (mathewc) [inline src/WebJobs.Script/Description/NodeFunctionInvoker.cs]: No reason really.

**Pull request closed without merging**